### DISCUSSION TITLE
Deployment error with Dynamically loaded JS

### DISCUSSION THREAD

**Discussion by kwoeste11** [CropperBlazor/Cropper.Blazor#312] [General]
Got everything working as I want, but when deploying to an IIS windows server, I am having issues with the dynamic JS in the cropper.   This is .net 8 and I just installed it on the server.  Everything else in the app works correctly.   Console errors below.   Hoping maybe you have run into this before and can shed some insight.  could well be a server config issue.  

![image](https://github.com/CropperBlazor/Cropper.Blazor/assets/144149160/7751e520-41bd-49dd-8a3f-77eabe358639)

**Comment by MaxymGorn** [comment]
Hi @kwoeste11. Previously we have this problem with loading cropper js interop module in this conversation: https://github.com/CropperBlazor/Cropper.Blazor/issues/142#issuecomment-1603316006. As result, you should override internal or full global path to folder: 
<img width="696" alt="image" src="https://github.com/CropperBlazor/Cropper.Blazor/assets/50423072/49509109-359d-4afd-8578-611e541084a4">
<img width="667" alt="image" src="https://github.com/CropperBlazor/Cropper.Blazor/assets/50423072/41e48c69-bd69-4b42-8994-e3ac4d2981d8">
Docs: https://cropperblazor.github.io/
 
You can see "cropperJsInterop.min.js" file via dev tools: 
<img width="1121" alt="image" src="https://github.com/CropperBlazor/Cropper.Blazor/assets/50423072/2619d65a-5734-4501-b035-4901f1c25dd2">

**Comment by kwoeste11** [comment]
Thanks again for your prompt help.  That was the issue.  When I originally read that in the notes, I didn't read it as the time it would be needed was when running under a virtual directory.   In my case this site runs under a virtual directory of "ImageEditor".   The code below allows debugging in the IDE as well as setting the path I need if it is running on the server.   Perhaps it helps someone else.

```
// Override internal path to CropperJSInterop.min.js module.  The server runs under a virtual
// directory and that needs to be part of the path.  For debugging in the IDE you will need to
// use the version without the ImageEdtior as part of the path
if (!Debugger.IsAttached)
{
	builder.Services.AddCropper(new CropperJsInteropOptions()
	{
		DefaultInternalPathToCropperModule = "ImageEditor/_content/Cropper.Blazor/cropperJsInterop.min.js"
	});
}
else
{
	builder.Services.AddCropper();
}
```
